iothub/service/src/Query/QueryResponse.cs

```diff
@@ -3,194 +3,79 @@
-                // Current page of results still had an item to return to the user.
-                Current = _items.Current;
-                Debug.Assert(_items.Current != null);
-                return true;
-            }
+        public override void Dispose()
+        {
+            _httpResponse?.Dispose();
+        }
 
-            if (ContinuationToken == null && queryOptions?.ContinuationToken == null)
-            {
-                // The query has no more pages of results to return and the last page has been
-                // completely exhausted.
-                return false;
-            }
+        protected override bool ContainsHeader(string name)
+        {
+            return _httpResponse.Headers.Contains(name);
+        }
 
-            // User's can pass in a continuation token themselves, but the default behavior
-            // is to use the continuation token saved by this class when it last retrieved a page.
-            var queryOptionsClone = new JobQueryOptions
-            {
-                ContinuationToken = queryOptions?.ContinuationToken ?? ContinuationToken,
-                PageSize = queryOptions?.PageSize ?? _defaultPageSize,
-                JobType = _jobType,
-                JobStatus = _jobStatus,
-            };
+        protected override IEnumerable<HttpHeader> EnumerateHeaders()
+        {
+            return _httpHeaders;
+        }
 
-            if (!string.IsNullOrEmpty(_originalQuery))
+        protected override bool TryGetHeader(string name, out string value)
+        {
+            IEnumerable<string> outVariableHeaders = new List<string>();
+            bool found = _httpResponse.Headers.TryGetValues(name, out outVariableHeaders);
+            if (found)
             {
-                QueryResponse<T> response = await _client
-                    .CreateAsync<T>(_originalQuery, queryOptionsClone, cancellationToken)
-                    .ConfigureAwait(false);
-                CurrentPage = response.CurrentPage;
-                _items = CurrentPage.GetEnumerator();
-                _items.MoveNext();
-                Current = _items.Current;
-                ContinuationToken = response.ContinuationToken;
+                value = outVariableHeaders.First();
             }
             else
             {
-                // Job type and job status may still be null here, but that's okay
-                QueryResponse<ScheduledJob> response = await _client
-                    .CreateJobsQueryAsync(queryOptionsClone, cancellationToken)
-                    .ConfigureAwait(false);
-                CurrentPage = (IEnumerable<T>)response.CurrentPage;
-                Current = CurrentPage.GetEnumerator().Current;
-                ContinuationToken = response.ContinuationToken;
+                value = "";
             }
 
-            return true;
+            return found;
+        }
+
+        protected override bool TryGetHeaderValues(string name, out IEnumerable<string> values)
```

Comment:
See `HttpHeadersExtensions.SafeGetValue)`

Reply:
Nice

Reply:
Does this method still have callers?